stdlib/public/core/Diffing.swift

```diff
@@ -0,0 +1,805 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2015 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// MARK: Diff application to RangeReplaceableCollection
+
+@available(macOS 9999, iOS 9999, tvOS 9999, watchOS 9999, *) // FIXME(availability-5.1)
+extension CollectionDifference {
+  fileprivate func _fastEnumeratedApply(
```

Comment:
At some point only `internal` and `public` declarations were allowed in the stdlib. I wonder what happened to that rule. @lorentey, @milseman do you know?

Reply:
I believe that private and fileprivate can work (but not as usableFromInline) in a world where we're not sil-serialize-all

Reply:
We started to occasionally use private in the stdlib last year. It's still a bit unusual, and the fileprivate/private mess has annoying interactions with our rule for explicit access levels. But as far as I know, the technical issues we had before have all been resolved now.